spec/views/shared/_banner.html.erb_spec.rb

```diff
@@ -1,6 +1,21 @@
 require 'rails_helper'
 
 describe 'shared/_banner.html.erb' do
+  before do
+    sp_with_logo = build_stubbed(
+      :service_provider, logo: 'generic.svg', friendly_name: 'Best SP ever'
+    )
+
+    decorated_session = ServiceProviderSessionDecorator.new(
+      sp: sp_with_logo,
+      view_context: '',
+      sp_session: {},
+      service_provider_request: nil,
+    )
+    allow(view).to receive(:decorated_session).and_return(decorated_session)
+    render
```

Comment:
I'd suggest removing this render before merging. Doing this in a before action means you'll actually render the view before any of the `it` block have a chance to do any kind of setup. I _think_ the results will be wiped out on a new render call, so things should still work, but you are doing an unnecessary render and it could be confusing for future travelers in this bit of code.